math/mathcore/inc/Math/CladDerivator.h

```diff
@@ -659,6 +677,490 @@ inline void landau_cdf_pullback(double x, double xi, double x0, double d_out, do
    *d_xi += _d_v * -((x - x0) / (xi * xi));
 }
 
+#ifdef R__HAS_MATHMORE
+
+inline void inc_gamma_c_pullback(double a, double x, double _d_y, double *_d_a, double *_d_x);
+
+inline void inc_gamma_pullback(double a, double x, double _d_y, double *_d_a, double *_d_x)
```

Comment:
Would that chatgpt suggestion work:

```cpp
inline void inc_gamma_pullback(double a, double x, double _d_y, double *_d_a, double *_d_x) {
    // Constants
    constexpr double kMACHEP = 1.11022302462515654042363166809e-16;
    constexpr double kMAXLOG = 709.782712893383973096206318587;
    
    double _d_ans = 0, _d_ax = 0, _d_c = 0, _d_r = 0;
    bool cond_a_nonpositive = (a <= 0);
    bool cond_x_nonpositive = (x <= 0);
    bool cond_large_x_a = (x > 1.0) && (x > a);

    if (cond_a_nonpositive || cond_x_nonpositive) return;

    if (cond_large_x_a) {
        double _r_a = 0, _r_x = 0;
        inc_gamma_c_pullback(a, x, -_d_y, &_r_a, &_r_x);
        *_d_a += _r_a;
        *_d_x += _r_x;
        return;
    }

    double ax = a * std::log(x) - x - std::lgamma(a);
    if (ax < -kMAXLOG) return;

    ax = std::exp(ax);
    double r = a, c = 1.0, ans = 1.0;
    unsigned long iterations = 0;

    do {
        iterations++;
        r += 1.0;
        c *= x / r;
        ans += c;
    } while (c / ans > kMACHEP);

    // Apply derivatives on the accumulated results
    _d_ans += _d_y / a * ax;
    _d_ax += ans * _d_y / a;
    *_d_a += _d_y * -(ans * ax / (a * a));

    while (iterations--) {
        ans -= c;
        _d_c += _d_ans;
        
        double temp_c = c;
        c /= (x / r);
        *_d_x += temp_c * _d_c / r;
        _d_r += temp_c * _d_c * -(x / (r * r));
        r -= 1.0;
    }

    *_d_a += _d_r;
    _d_ax += std::exp(ax) * _d_ax;
    
    ax -= a * std::log(x);
    *_d_a += _d_ax * std::log(x) - _d_ax * std::lgamma(a);
    *_d_x += _d_ax * (a / x) - _d_ax;
}
```

Reply:
I'll check!

Reply:
No wait, it can't. the `digamma` doesn't event appear there anymore. The code can't be simplified like that, getting rid of an important inner derivative. I have however applied some manual improvements to the code now, in particular to get rid of `goto` statements.